gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-oidc/src/test/java/io/gravitee/am/gateway/handler/oidc/service/idtoken/IDTokenServiceTest.java

```diff
@@ -220,6 +221,72 @@ public void shouldCreateIDToken_customClaims() {
+        verify(certificateManager, times(1)).defaultCertificateProvider();
+        verify(jwtService, times(1)).encode(any(), eq(defaultCert));
+    }
+
+    @Test
+    public void shouldCreateIDToken_customClaims_with_exisiting_user() {
+        OAuth2Request oAuth2Request = new OAuth2Request();
+        oAuth2Request.setClientId("client-id");
+        oAuth2Request.setScopes(Set.of("openid", "full_profile"));
+
+        TokenClaim customClaim = new TokenClaim();
+        customClaim.setTokenType(TokenTypeHint.ID_TOKEN);
+        customClaim.setClaimName("iss");
+        customClaim.setClaimValue("https://custom-iss");
+
+        Client client = new Client();
+        client.setCertificate("certificate-client");
+        client.setClientId("my-client-id");
+        client.setTokenCustomClaims(Arrays.asList(customClaim));
+
+        ExecutionContext executionContext = mock(ExecutionContext.class);
+        TemplateEngine templateEngine = mock(TemplateEngine.class);
+        when(templateEngine.getValue("https://custom-iss", Object.class)).thenReturn("https://custom-iss");
+        when(executionContext.getTemplateEngine()).thenReturn(templateEngine);
+
+        String idTokenPayload = "payload";
+        io.gravitee.am.gateway.certificate.CertificateProvider defaultCert = new io.gravitee.am.gateway.certificate.CertificateProvider(defaultCertificateProvider);
+
+        ArgumentCaptor<JWT> jwtCaptor = ArgumentCaptor.forClass(JWT.class);
+        when(jwtService.encode(jwtCaptor.capture(), any(io.gravitee.am.gateway.certificate.CertificateProvider.class))).thenReturn(Single.just(idTokenPayload));
+        when(certificateManager.findByAlgorithm(any())).thenReturn(Maybe.empty());
+        when(certificateManager.get(any())).thenReturn(Maybe.empty());
+        when(certificateManager.defaultCertificateProvider()).thenReturn(defaultCert);
+
+        var user = new User();
+        user.setId(UUID.randomUUID().toString());
+        user.setAdditionalInformation(
+            Map.of(
+                "userClaim", "Some that will exist",
+                Claims.nbf, "Some value that won't exist",
+                Claims.iat, "Some value that won't exist",
+                Claims.exp, "Some value that won't exist",
+                ConstantKeys.OIDC_PROVIDER_ID_ACCESS_TOKEN_KEY, "Some value that won't exist",
+                ConstantKeys.OIDC_PROVIDER_ID_TOKEN_KEY, "Some value that won't exist"
+            )
+        );
+        oAuth2Request.setSubject(user.getId());
+
+        TestObserver<String> testObserver = idTokenService.create(oAuth2Request, client, user, executionContext).test();
+        testObserver.assertComplete();
+        testObserver.assertNoErrors();
+
+        JWT jwt = jwtCaptor.getValue();
+
+        assertNotNull(jwt);
+        assertEquals("https://custom-iss", jwt.get("iss"));
+        assertEquals("Some that will exist", jwt.get("userClaim"));
+        assertNotEquals("Some value that won't exist", jwt.get(Claims.nbf));
```

Comment:
I think that the idea is that the value is not present. Then maybe the assertion can check for `null` instead of asserting a non existing value

Reply:
The value exists but is not the one we expect